eng/scripts/validate_go_mod.ps1

```diff
@@ -7,17 +7,19 @@ Push-Location sdk/$serviceDir
 $goModFiles = Get-ChildItem -Path . -Filter go.mod -Recurse
 
 if ($goModFiles.Length -eq 0) {
-    Write-Host "Could not find a go.mod file in the directory $(pwd)"
+    Write-Host "Could not find a go.mod file in the directory $(Get-Location)"
     exit 1
 }
 
 $hasError = $false
 foreach ($goMod in $goModFiles) {
-    $name = $goMod.FullName
-    $patternMatches = Get-Content $name | Select-String -Pattern "replace "
-    if ($patternMatches.Length -ne 0) {
-        Write-Host "Found a replace directive in go.mod file at $name"
-        $hasError = $true
+    if (-Not ($goMod.FullName -Like "*testdata*")) {
```

Comment:
I'd suggest using the common https://github.com/Azure/azure-sdk-for-go/blob/main/eng/scripts/Language-Settings.ps1#L37 they already return null for these non-standard go.mod files.